ample/ensembler/ensemble_rmsds.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env ccp4-python
+import os
+import shutil
+import numpy as np
+import iotbx.pdb
+from pyjob import Job
+
+def ensemble_rmsds(ensemble, gesamt_exe):
+    """Quick attempt to add PHASER RMSD commands to ensemble
+    Currently just for testing"""
+    
+    def parse_rmsd_pair(fh):
+        for line in fh:
+            line = line.strip()
+            if not line:
+                continue
+            if 'RMSD             :' in line:
+                rmsd = float(line.split()[2])
+                break
+        assert rmsd
+        return [rmsd / 2.0] * 2
+
+    def parse_rmsd_multi(fh):
+        mlist = []
+        reading = 0
+        for line in fh:
+            line = line.strip()
+            if not line:
+                continue
+            if '____' in line and reading == 2:
+                break
+            if '----' in line and reading == 1:
+                reading = 2
+                continue
+            if '(o) pairwise r.m.s.d. (consensus r.m.s.d. on diagonal):' in line:
+                reading = 1
+                continue
+            if reading == 2:
+                mlist.append([float(r) for r in line.split()[1:]])
+        return list(np.array(mlist).diagonal())
+
+    nmodels = iotbx.pdb.pdb_input(ensemble).construct_hierarchy().models_size()
+     
+    cmd = '{} '.format(gesamt_exe)
+    for i in range(nmodels):
+        cmd += '{} -s "/{}/" '.format(ensemble, i+1)
+     
+    script = './gesamt.sh'
+    logfile = 'gesamt.log'
+    with open(script, 'w') as w:
+        w.write('#!/bin/bash\n{}\n\n'.format(cmd))
```

Comment:
Also, I see you're writing a script. How about the pyjob [`make_script`](https://github.com/fsimkovic/pyjob/blob/ed80d574abf4924c2fbb84e57a43fe1075b8014d/pyjob/misc/__init__.py#L46) function. Might be handy as it performs all relevant OS-dependent checks.

Reply:
Commands are provided as list of lists, i.e. list of lines.

Reply:
Ok, good to know.